subsys/bluetooth/host/l2cap.c

```diff
@@ -82,6 +82,14 @@ K_FIFO_DEFINE(free_l2cap_tx_meta_data);
 
 static struct l2cap_tx_meta_data *alloc_tx_meta_data(void)
 {
+	/* The meta data always get freed in the system workqueue,
+	 * so if we're in the same workqueue but there are no immediate
+	 * contexts available, there's no chance we'll get one by waiting.
+	 */
+	if (k_current_get() == &k_sys_work_q.thread) {
+		return k_fifo_get(&free_l2cap_tx_meta_data, K_NO_WAIT);
```

Comment:
Isn't this line the exact same as line 93?

Reply:
copy-paste!